File: 169. Majority Element/Main.java
// https://leetcode.com/problems/majority-element/submissions/1079590835/
class Solution {
  public int majorityElement(int[] nums) {
    Map<Integer, Integer> majorNum = new HashMap<>();
    for (int x = 0; x < nums.length; x++) {
      if (majorNum.containsKey(nums[x])) {
        majorNum.put(nums[x], majorNum.get(nums[x]) + 1);
      } else {
        majorNum.put(nums[x], 1);
      }
    }
    int prevKey = 0;
    int prevValue = 0;
    for (Map.Entry<Integer, Integer> entry : majorNum.entrySet()) {
      if (entry.getValue() >= prevValue) {
        prevValue = entry.getValue();
        prevKey = entry.getKey();
      }
    }
    return prevKey;
  }
}

File: 169. Majority Element/main.cpp
/*
https://leetcode.com/problems/majority-element/submissions/1079590835/
*/
class Solution {
public:
  int majorityElement(vector<int> &nums) {
    map<int, int> majorNum;
    for (int x = 0; x < nums.size(); x++) {
      majorNum[nums[x]] = 0;
    }
    for (int x = 0; x < nums.size(); x++) {
      if (majorNum.find(nums[x])->first == nums[x]) {
        majorNum[nums[x]]++;
      }
    }
    for (int x = 0; x < nums.size(); x++) {
      cout << majorNum.find(nums[x])->first << " "
           << majorNum.find(nums[x])->second << endl;
    }
    int aux = 0, val = 0;
    for (int x = 0; x < nums.size(); x++) {
      // cout << "a: " << aux << "\n";
      cout << "a: " << aux << " " << majorNum.find(nums[x])->second << "\n";
      if (majorNum.find(nums[x])->second >= aux) {
        aux = majorNum.find(nums[x])->second;
        val = majorNum.find(nums[x])->first;
      }
    }

    return val;
  }
};